crates/sui-package-dump/src/lib.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::BTreeMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{bail, ensure, Context, Result};
+use client::Client;
+use fastcrypto::encoding::{Base64, Encoding};
+use query::{limits, packages, SuiAddress, UInt53};
+use sui_types::object::Object;
+use tracing::info;
+
+mod client;
+mod query;
+
+/// Ensure all packages created before `before_checkpoint` are written to the `output_dir`ectory,
+/// from the GraphQL service at `rpc_url`.
+///
+/// `output_dir` can be a path to a non-existent directory, in which case it will be created, or an
```

Comment:
```suggestion
/// `output_dir` can be a path to a non-existent directory, an existing empty directory, or an existing directory written to in the past. If the path is non-existent, the invocation creates it. If the path exists but is empty, the invocation writes to the directory. If the directory has been written to in the past, the invocation picks back up where the previous invocation left off.
```
This sentence is way long as is. Suggest breaking it up.